### PR TITLE
Deletes TAs after expiration 

### PR DESCRIPTION
**Pull request by tnickelsen**
and corrected activity log entries for proposals

### CODE REVIEW

File: domains/transfer/Transfer.API/API/Transfer/TransferAgreementProposalCleanup/TransferAgreementProposalCleanupService.cs [hunk 1]
```diff
@@ -53,7 +53,7 @@ private async Task DeleteOldTransferAgreementProposals(DateTimeOffset olderThan,
         context.TransferAgreementProposals.RemoveRange(oldProposals);
 
         var activityLogEntries = oldProposals.Select(proposal => ActivityLogEntry.Create(Guid.Empty, ActivityLogEntry.ActorTypeEnum.System,
-            string.Empty, string.Empty, string.Empty, ActivityLogEntry.EntityTypeEnum.TransferAgreementProposal,
+            string.Empty, proposal.SenderCompanyTin, proposal.SenderCompanyName, ActivityLogEntry.EntityTypeEnum.TransferAgreementProposal,
```
Comment (sahma19) [inline domains/transfer/Transfer.API/API/Transfer/TransferAgreementProposalCleanup/TransferAgreementProposalCleanupService.cs:56]: Why this change?

File: domains/transfer/Transfer.API/API/appsettings.Development.json [hunk 2]
```diff
@@ -21,6 +21,9 @@
     "TransferAgreementProposalCleanupService": {
         "SleepTime": "01:00:00"
     },
+    "TransferAgreementCleanup": {
+        "SleepTime": "01:00:00"
+    },
```
Comment (sahma19) [inline domains/transfer/Transfer.API/API/appsettings.Development.json:26]: So right now we have multiple workers starting every hour. will that be a potential problem now, or in the future?

File: domains/transfer/Transfer.API/API.IntegrationTests/Transfer/TransferAgreementCleanup/TransferAgreementCleanupTests.cs [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+            Id = Guid.NewGuid(),
+            EndDate = DateTimeOffset.UtcNow.AddHours(-1),
+            ReceiverTin = receiverTin,
+            SenderName = "SomeSender",
+            SenderTin = tin,
+            ReceiverReference = Guid.NewGuid(),
+            SenderId = sub,
+            StartDate = DateTimeOffset.UtcNow.AddDays(-1),
+            TransferAgreementNumber = 0
+        };
```
Comment (sahma19) [inline domains/transfer/Transfer.API/API.IntegrationTests/Transfer/TransferAgreementCleanup/TransferAgreementCleanupTests.cs:111]: Is this a valid TA? start date must be before enddate

File: domains/transfer/Transfer.API/API.IntegrationTests/Transfer/TransferAgreementCleanup/TransferAgreementCleanupTests.cs [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+        var expiredTa = new TransferAgreement
+        {
+            Id = Guid.NewGuid(),
+            EndDate = DateTimeOffset.UtcNow.AddHours(-1),
+            ReceiverTin = "12345678",
+            SenderName = "SomeSender",
+            SenderTin = tin,
+            ReceiverReference = Guid.NewGuid(),
+            SenderId = sub,
+            StartDate = DateTimeOffset.UtcNow.AddDays(-1),
```
Comment (sahma19) [inline domains/transfer/Transfer.API/API.IntegrationTests/Transfer/TransferAgreementCleanup/TransferAgreementCleanupTests.cs:49]: StartDate before enddate plz

File: domains/transfer/Transfer.API/API.IntegrationTests/Transfer/TransferAgreementCleanup/TransferAgreementCleanupTests.cs [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+        var nullEndDateTa = new TransferAgreement
+        {
+            Id = Guid.NewGuid(),
+            EndDate = null,
+            ReceiverTin = "12345679",
+            SenderName = "SomeSender",
+            SenderTin = tin,
+            ReceiverReference = Guid.NewGuid(),
+            SenderId = sub,
+            StartDate = DateTimeOffset.UtcNow.AddDays(-1),
```
Comment (sahma19) [inline domains/transfer/Transfer.API/API.IntegrationTests/Transfer/TransferAgreementCleanup/TransferAgreementCleanupTests.cs:61]: StartDate before enddate plz

File: domains/transfer/Transfer.API/API.IntegrationTests/Transfer/TransferAgreementCleanup/TransferAgreementCleanupTests.cs [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+        var endDateTa = new TransferAgreement
+        {
+            Id = Guid.NewGuid(),
+            EndDate = DateTimeOffset.UtcNow.AddHours(1),
+            ReceiverTin = "12345677",
+            SenderName = "SomeSender",
+            SenderTin = tin,
+            ReceiverReference = Guid.NewGuid(),
+            SenderId = sub,
+            StartDate = DateTimeOffset.UtcNow.AddDays(-1),
```
Comment (sahma19) [inline domains/transfer/Transfer.API/API.IntegrationTests/Transfer/TransferAgreementCleanup/TransferAgreementCleanupTests.cs:73]: StartDate before enddate plz

File: domains/transfer/Transfer.API/API.IntegrationTests/Transfer/TransferAgreementCleanup/TransferAgreementCleanupTests.cs [hunk 7]
```diff
@@ -0,0 +1,135 @@
...
+        var expiredTa = new TransferAgreement
+        {
+            Id = Guid.NewGuid(),
+            EndDate = DateTimeOffset.UtcNow.AddHours(-1),
+            ReceiverTin = receiverTin,
+            SenderName = "SomeSender",
+            SenderTin = tin,
+            ReceiverReference = Guid.NewGuid(),
+            SenderId = sub,
+            StartDate = DateTimeOffset.UtcNow.AddDays(-1),
```
Comment (sahma19) [inline domains/transfer/Transfer.API/API.IntegrationTests/Transfer/TransferAgreementCleanup/TransferAgreementCleanupTests.cs:109]: StartDate before enddate plz

File: domains/transfer/Transfer.API/API/Transfer/TransferAgreementCleanup/Options/TransferAgreementCleanupOptions.cs [hunk 8]
```diff
@@ -0,0 +1,13 @@
+using System.ComponentModel.DataAnnotations;
+using System;
+
+namespace API.Transfer.TransferAgreementCleanup.Options;
+
+public class TransferAgreementCleanupOptions
+{
+    public const string Prefix = "TransferAgreementCleanup";
+
+    [Range(typeof(TimeSpan), "00:00:02", "02:00:00")]
```
Comment (sahma19) [inline domains/transfer/Transfer.API/API/Transfer/TransferAgreementCleanup/Options/TransferAgreementCleanupOptions.cs:10]: I thought it was every hour, and not every 2 hours?

**Review by sahma19 — CHANGES_REQUESTED** [review]

File: domains/transfer/Transfer.API/API/Transfer/TransferAgreementProposalCleanup/TransferAgreementProposalCleanupService.cs (hunk 1 above)
Comment (tnickelsen) [reply domains/transfer/Transfer.API/API/Transfer/TransferAgreementProposalCleanup/TransferAgreementProposalCleanupService.cs:56]: This is a fix. Now we can actually see that the proposal has been deleted in the activity log. We could now before since the orgTin was string empty

Comment (sahma19) [reply domains/transfer/Transfer.API/API/Transfer/TransferAgreementProposalCleanup/TransferAgreementProposalCleanupService.cs:56]: Yes, but how will the message look like then? Will it look like as if the senderCompanyName deleted the TA?

File: domains/transfer/Transfer.API/API.IntegrationTests/Transfer/TransferAgreementCleanup/TransferAgreementCleanupTests.cs (hunk 3 above)
Comment (tnickelsen) [reply domains/transfer/Transfer.API/API.IntegrationTests/Transfer/TransferAgreementCleanup/TransferAgreementCleanupTests.cs:111]: StartDate is before EndDate? It says AddDays :)

File: domains/transfer/Transfer.API/API/Transfer/TransferAgreementCleanup/Options/TransferAgreementCleanupOptions.cs (hunk 8 above)
Comment (tnickelsen) [reply domains/transfer/Transfer.API/API/Transfer/TransferAgreementCleanup/Options/TransferAgreementCleanupOptions.cs:10]: We can configure it to every 2 hours if we want but this will be configured to every hour.

File: domains/transfer/Transfer.API/API.IntegrationTests/Transfer/TransferAgreementCleanup/TransferAgreementCleanupTests.cs (hunk 3 above)
Comment (sahma19) [reply domains/transfer/Transfer.API/API.IntegrationTests/Transfer/TransferAgreementCleanup/TransferAgreementCleanupTests.cs:111]: ooopppppps

File: domains/transfer/Transfer.API/API/Transfer/TransferAgreementProposalCleanup/TransferAgreementProposalCleanupService.cs (hunk 1 above)
Comment (tnickelsen) [reply domains/transfer/Transfer.API/API/Transfer/TransferAgreementProposalCleanup/TransferAgreementProposalCleanupService.cs:56]: Gonna check in vcluster. Asked Martin and he said I should put paramters like this

Comment (tnickelsen) [reply domains/transfer/Transfer.API/API/Transfer/TransferAgreementProposalCleanup/TransferAgreementProposalCleanupService.cs:56]: Or actually this I can't test in vCluster since it takes 14 days to delete a proposal. I'm gonna trust Martin - the paramters are called organizationName and organizationTin so they must be related. We'll change it if it looks wierd. Ok with you?

Comment (sahma19) [reply domains/transfer/Transfer.API/API/Transfer/TransferAgreementProposalCleanup/TransferAgreementProposalCleanupService.cs:56]: Make an integration test and check? See the output when you make a get request to activity log endpoint

Comment (sahma19) [reply domains/transfer/Transfer.API/API/Transfer/TransferAgreementProposalCleanup/TransferAgreementProposalCleanupService.cs:56]: What I am asking is. If worker is deleting a proposal, should it then be logged in an activity log, as if it was organizationName and OrganizationTin that did it?

Comment (tnickelsen) [reply domains/transfer/Transfer.API/API/Transfer/TransferAgreementProposalCleanup/TransferAgreementProposalCleanupService.cs:56]: If you see the code I have an integration test for this. I just can't see how the frontend interprets it

**Review by sahma19 — APPROVED** [review]
